### PR TITLE
 	Scenes for Camp Sieges, Sorcerer Quest, Osgiliath landmark, stronger destructible gates, new props

### PR DESCRIPTION
**Pull request by In-Vain**
(Trying to get the hang of doing proper pull requests and github in general before I start pushing changes myself. You see I had to revert.)

-Prettified Orc Sentry Camp, Uruk-hai Outpost, Uruk-hai River Camp, Goblin South Outpost and made sieges a bit more interesting
- made all destructible gates stronger
- 3 new scenes for the sorcerer quest
- 4 new scenes for Osgiliath landmark battles
- some new scene props: Mirkwood trees, destructible gate for Orc camps, dolmen stones

### CODE REVIEW

File: ModuleSystem/module_scene_props.py [hunk 1]
```diff
@@ -2535,6 +2535,62 @@
...
 	( "boat_sail_off"                              ,0,"boat_sail_off","0",[]),
 
+# from '': begin (OpenBRF)
+	( "tree_mirkwood_a"                            ,0,"tree_mirkwood_a","bo_tree_mallorn_a",[]),
+	( "tree_mirkwood_b"                            ,0,"tree_mirkwood_b","bo_tree_mallorn_b",[]),
+	( "tree_mirkwood_c"                            ,0,"tree_mirkwood_c","bo_tree_mallorn_c",[]),
+	( "tree_mirkwood_roots_1"                      ,0,"tree_mirkwood_roots_1","0",[]),
+	( "tree_mirkwood_roots_2"                      ,0,"tree_mirkwood_roots_2","0",[]),
+	( "tree_mirkwood_roots_3"                      ,0,"tree_mirkwood_roots_3","0",[]),
+	( "tree_mirkwood_roots_4"                      ,0,"tree_mirkwood_roots_4","0",[]),
```
Comment (Swyter) [inline ModuleSystem/module_scene_props.py:2545]: @In-Vain This OpenBRF formatting is a bit awkward, but will do. Try to keep the simple prop per line rule to make it more readable. Complex things with triggers like the one below are fine, of course.

File: ModuleSystem/ID/ID_scene_props_mb.py [hunk 2]
```diff
@@ -1554,32 +1554,23 @@
...
+spr_save_compartibility11 = 1566
+spr_save_compartibility12 = 1567
+spr_save_compartibility13 = 1568
+spr_save_compartibility14 = 1569
+spr_save_compartibility15 = 1570
+spr_save_compartibility16 = 1571
+spr_save_compartibility17 = 1572
+spr_save_compartibility18 = 1573
+spr_save_compartibility19 = 1574
+spr_save_compartibility20 = 1575
```
Comment (Swyter) [inline ModuleSystem/ID/ID_scene_props_mb.py]: @In-Vain This kind of back-and-forth can be avoided, specially when you are preparing your changes to make them public. Read this:
https://www.atlassian.com/git/tutorials/rewriting-history

And if you want something a bit riskier try this:
https://www.atlassian.com/git/tutorials/undoing-changes

Keep in mind that undoing changes in the common repository is better left for when you know what you are doing, you can safely play around in your own personal copies/clones, though. Like the one containing the changes I'm reviewing.

**Comment by Swyter** [comment]
@In-Vain Finally had some time to review this. Take a look at https://github.com/tldmod/tldmod/pull/12#discussion_r125516693 and see if you can cleanly remove the first two or three commits from your repository, leaving only the interesting stuff.

If not I can always merge it like this, but keep that in mind for future changes. If you rewrite your branch history it will appear here. No need to close and create a new pull request.

File: ModuleSystem/ID/ID_scene_props_mb.py (hunk 2 above)
Comment (In-Vain) [reply ModuleSystem/ID/ID_scene_props_mb.py]: Thanks! I'll have a look and try to clean up the pull request. Probably not before next week, but there's no need to hurry merging it.
What's wrong with that specific part from ID_scene_props you quoted?

Comment (Swyter) [reply ModuleSystem/ID/ID_scene_props_mb.py]: @In-Vain Looking at it again I can see that those weren't carriage returns, but a lot of space filling.

**Comment by In-Vain** [comment]
@Swyter Uh, seems like I screwed up this one. I managed to "hide" the merge, first commit and revert commit by resetting and it looks clean on github desktop, but syncing created yet another merge and also it seems that all the unneeded commits are still included in this pull request.

Am I missing some basics?

**Comment by Swyter** [comment]
@In-Vain I know that Git is a pain in places, but what I would do is to save https://github.com/tldmod/tldmod/pull/12/commits/31e389eef6df987e399eb60ebbb636b183eeec6d as [its own patch][1] (ensure you have a complete copy to avoid losing your work; here's a full patch of your commit saved for posterity: http://archive.is/LkUp3) and then nuke all the commits you have added by [using this handy guide][2].

Once the branch is pristine you can reapply the single patch (the only containing meaningful changes) and it should appear here ready to merge. If you manage to do this you will have mastered most of what you need to know about Git.

Personally, I hate the GitHub client and its hipster UI; I believe that it makes everything more complicated and confusing than what it really is. Use the terminal if you can, it pays off in the long run.

[1]: https://stackoverflow.com/questions/6658313/generate-a-git-patch-for-a-specific-commit
[2]: https://stackoverflow.com/a/1338744/674685

**Comment by Swyter** [comment]
@In-Vain Do you want me to commit it for you?

**Comment by In-Vain** [comment]
Hold on, I'm in the middle of figuring it out ;)
I guess the request got closed automatically when I reset my repository. Now for applying that patch...

**Comment by In-Vain** [comment]
There we go! Thanks for your patience! :)

**Comment by In-Vain** [comment]
Dammit, apparently not. Apparently I forgot to sync before applying the patch...

Please bear with me!

**Comment by In-Vain** [comment]
This should be it. Sorry for all the hassle. The last commit being not compiled confused the hell out of me.

**Comment by Swyter** [comment]
@In-Vain No worries. I see that @khamukkamu has made a change while you were doing your thing (why oh why? :)), so it also appears on top of your pull request because you updated your local copy and then pushed (meaning that if we merge this we'll have two repeated commits). Do you want to rebase it or are you already getting war flashbacks from your recent Git battles?

You can also squash (collapse) your change and your version of Kham's change together by using the arrow of the bit _Merge pull request_ button right from this page.

    Legend: (a) is the base (b) is your change, (c) is the recent Kham change,
            (d) is the future commit merging all that stuff in your branch with the main thing.

            We would get two (c) commits.

         (a)       (c) (d)
    ...---+---------+---+--->
           \___+____+__/
    
              (b)  (c)

By rebasing we would get the ideal thing:

         (a) (c)      (d)
    ...---+---+--------+--->
              \___+___/
    
                 (b)   



___Edit___: Actually, just do a `git reset --hard HEAD` in your `In-Vain:master` branch to delete your version of `(c)`. That way we can rebase it from here. That's easy and way better.


Next time just open a new branch for every pull request instead of using your main (`master`) branch, that's simplifies stuff, and let's you have your `master` branch up-to-date without polluting your unsent changes.

**Comment by In-Vain** [comment]
Oh dear, and I was so eager to make sure to base my commit on Khamu's |-)

I understand what happened, but I don't quite get what to reset to. If I `git reset --hard HEAD`, nothing happens, because my own commit, already including Khamu's, is the last one. If I hard reset to Khamu's commit, I lose mine, right?

**Comment by Swyter** [comment]
@In-Vain The simpler way is to reset everything (let's say by removing the last four commits), then doing a `git pull` to re-grab the latest stuff from upstream and then commit your patch. That is like a primitive rebase, but should be doable.

Don't sync after that, at least until this thing gets merged, or new commits from the main repo will appear in this pull request too.

----

The problem right now it that you are running a parallel/cloned history, but your branch should only have your own changes since that point in the common history.

**Comment by In-Vain** [comment]
@Swyter Sorry for causing all that trouble! :(
However, thanks to you, I learned quite a bit today. I'll look into creating new branches for commits from now on. This works by doing `git push origin <name of the new branch>`, doesn't it?

**Comment by Swyter** [comment]
@In-Vain Everything's good now! (Even if that beautiful commit description is gone now: https://github.com/tldmod/tldmod/commit/31e389eef6df987e399eb60ebbb636b183eeec6d)

If you want to know more about creating your own feature branches, read this:
https://git-scm.com/book/en/v1/Git-Branching-What-a-Branch-Is

It's pretty complete. Next time you open a pull request you will see that you can select from which branch you want to merge your changes.

Do you want to click on the merge button? Try doing a "_squash and merge_" (click on the downward arrow). That way you can change the commit description to something more complete right from the browser. Like pasting the previous text.

**Comment by In-Vain** [comment]
Done, thanks!

**Comment by Swyter** [comment]
@In-Vain Well. There you go! Now Travis is doing its thing:

https://travis-ci.org/tldmod/tldmod/builds/252190519

You will see the nightlies appear in a few minutes, unless something goes amiss.

**Comment by Swyter** [comment]
@In-Vain Well, you are now the proud owner of revision 2324. Congratulations!

https://tldmod.github.io/

**Pull request merged**